packages/core/src/utils/Error.ts

```diff
@@ -20,460 +20,504 @@ const {
+            return `${expectedType.args.length} arguments expected for type ${expectedType.name.value
+                } (defined at ${loc(expectedExpr)}), but ${sourceType.args.length
+                } arguments were given for ${sourceType.name.value} at ${loc(
+                    sourceExpr
+                )} `;
+        }
+        case "ArgLengthMismatch": {
+            const { name, argsGiven, argsExpected, sourceExpr, expectedExpr } = error;
+            return `${name.value} expects ${argsExpected.length
+                } arguments (originally defined at ${loc(expectedExpr)}), but was given ${argsGiven.length
+                } arguments instead at ${loc(sourceExpr)}.`;
+        }
+        case "DeconstructNonconstructor": {
+            const { variable, field } = error.deconstructor;
+            return `Becuase ${variable.value} is not bound to a constructor, ${variable.value
+                }.${field.value} (at ${loc(
+                    error.deconstructor
+                )}) does not correspond to a field value.`;
+        }
+        case "UnexpectedExprForNestedPred": {
+            const { sourceExpr, sourceType, expectedExpr } = error;
+            return `A nested predicate is expected (defined at ${loc(
+                expectedExpr
+            )}), but an expression of '${showType(
+                sourceType
+            )}' type was given at ${loc(sourceExpr)}.`;
+        }
+
+        // ---- BEGIN STYLE ERRORS
+        // COMBAK suggest improvements after reporting errors
+
+        case "GenericStyleError": {
+            return `DEBUG: Style failed with errors:\n ${error.messages.join("\n")}`;
+        }
+
+        case "SelectorDeclTypeError": {
+            // COMBAK Maybe this should be a TaggedSubstanceError?
+            return "Substance type error in declaration in selector";
+        }
+
+        case "StyleErrorList": {
+            return error.errors.map(showError).join(", ");
+        }
+
+        case "SelectorVarMultipleDecl": {
+            return "Style pattern statement has already declared the variable ${error.varName.value}";
+        }
+
+        case "SelectorDeclTypeMismatch": {
+            // COMBAK: Add code for prettyprinting types
+            return "Mismatched types or wrong subtypes between Substance and Style variables in selector";
+        }
+
+        case "SelectorRelTypeMismatch": {
+            // COMBAK: Add code for prettyprinting types
+            return "Mismatched types or wrong subtypes between variable and expression in relational statement in selector";
+        }
+
+        case "TaggedSubstanceError": {
+            return showError(error.error); // Substance error
+        }
+
+        // --- BEGIN BLOCK STATIC ERRORS
+
+        case "InvalidGPITypeError": {
+            return `Got invalid GPI type ${error.givenType.value}.`;
+            // COMBAK: GPI type suggestions, or just print the list
```

Comment:
I think printing out the list of GPI types is fine. Should be easy to do?

Reply:
Did this, and the properties too